app/controllers/application_controller.rb

```diff
@@ -13,7 +13,17 @@ def sort_methods
     ]
   end
 
-  def set_cart
-    @cart = Cart.new(session[:cart])
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id] && User.exists?(id: session[:user_id])
```

Comment:
Cool, like how you checked that the user exists too. Not sure if I've used that before.